dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -0,0 +1,372 @@
+            { AppNotificationSoundEvent::Call9, L"ms-winsoundevent:Notification.Looping.Call9"},
+            { AppNotificationSoundEvent::Call10, L"ms-winsoundevent:Notification.Looping.Call10"},
+        };
+
+        return c_soundEventMap[soundEvent];
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddArgument(hstring const& key, hstring const& value)
+    {
+        THROW_HR_IF(E_INVALIDARG, key.empty());
+
+        m_arguments.Insert(key, value);
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetTimeStamp(winrt::Windows::Foundation::DateTime const& value)
+    {
+        //winrt::hstring currentLanguage { ApplicationLanguages::Languages().First().Current() };
+        //// DateTimeFormatter dateTimeFormatter{ "longtime" };
+        //winrt::Windows::Foundation::DateTime::
+        //winrt::Windows::Foundation::DateTime { 2022, 1, 13, 16, 25, 30,  }
+        //m_timeStamp = timeStamp;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetScenario(AppNotificationScenario const& value)
+    {
+        m_scenario = value;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddText(hstring const& text)
+    {
+        THROW_HR_IF(E_INVALIDARG, m_textLines.size() == 3u);
+
+        m_textLines.push_back(L"<text>" + text + L"</text>");
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddText(hstring const& text, AppNotificationTextProperties const& properties)
+    {
+        THROW_HR_IF(E_INVALIDARG, m_textLines.size() == 3u);
+
+        m_textLines.push_back(properties.as<winrt::Windows::Foundation::IStringable>().ToString() + text + L"</text>");
+
+        if (properties.IncomingCallAlignment())
+        {
+            m_scenario = AppNotificationScenario::IncomingCall;
+        }
+        return *this;
+    }    
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetAttributionText(hstring const& text)
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_attributionText.empty());
+
+        m_attributionText = L"<text placement=\"attribution\">" + text + L"</text>";
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetAttributionText(hstring const& text, hstring const& language)
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_attributionText.empty());
+        THROW_HR_IF(E_INVALIDARG, language.empty());
+
+        m_attributionText = L"<text placement=\"attribution\" lang=\"" + language + L"\">" + text + L"</text>";
```

Comment:
General note: Use LR notation and avoid escape characters which are extremely difficult to read.